### ISSUE TITLE
Error [ERR_PACKAGE_PATH_NOT_EXPORTED] Package subpath './v4' is not defined by "exports"

### ISSUE THREAD

**Issue by peppescg** [coverallsapp/github-action#234]
I had unknown coverage so I tried to run locally coveralls with `./coverage/lcov.info`, but I got a node error related to exports not defined on a package.
Do you have any insights?
Node LTS  `v20.18.0`
____

```
node:internal/modules/cjs/loader:641
      throw e;
      ^

Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './v4' is not defined by "exports" in /Users/******/node_modules/uuid/package.json
    at exportsNotFound (node:internal/modules/esm/resolve:304:10)
    at packageExportsResolve (node:internal/modules/esm/resolve:651:9)
    at resolveExports (node:internal/modules/cjs/loader:634:36)
    at Module._findPath (node:internal/modules/cjs/loader:724:31)
    at Module._resolveFilename (node:internal/modules/cjs/loader:1211:27)
    at Module._load (node:internal/modules/cjs/loader:1051:27)
    at Module.require (node:internal/modules/cjs/loader:1311:19)
    at require (node:internal/modules/helpers:179:18)
    at Object.<anonymous> (/Users/******/node_modules/request/lib/auth.js:4:12)
    at Module._compile (node:internal/modules/cjs/loader:1469:14) {
  code: 'ERR_PACKAGE_PATH_NOT_EXPORTED'
}
```

**Comment by afinetooth** [comment]
@peppescg What [Coveralls Integration](https://docs.coveralls.io/integrations) are you using? The fact that you're posting in this forum suggests you're using our [Official Integration](https://docs.coveralls.io/integrations#official-integrations), the [Coveralls GitHub Action](https://github.com/marketplace/actions/coveralls-github-action), but that shouldn't be throwing a Node error, so are you also employing another integration, such as a [Language Integration](https://docs.coveralls.io/integrations#language-integrations) like [node-coveralls](https://docs.coveralls.io/javascript) (outdated)?

Or are you using `v1` of our GitHub Action, which uses [node-coveralls](https://github.com/nickmerwin/node-coveralls) under-the-hood?:

```
    - name: Coveralls
      uses: coverallsapp/github-action@v1
```

If so, I'd recommend using `@v2`:

```
    - name: Coveralls
      uses: coverallsapp/github-action@v2
```

Can you please share your config, your original response from the GitHub Action / POST to the Coveralls API, and any other details on your use case?

Can you also please supply the command you used to "run locally coveralls"?

Thanks.

**Comment by peppescg** [comment]
Hi @afinetooth , so I am using the [GH action](https://github.com/marketplace/actions/coveralls-github-action), pointed to the last commit. 
My issue is that I see from the actions that everything is ok, but I am not able to see the coverage in coveralls, and also I don't see it as PR comment. I enabled the PR comments feature through settings.

```
 name: Send coverage to Coveralls
        uses: coverallsapp/github-action@1134c89e4bf10443bf8f0ec69640b667cfd91041 #v2.3.0
        with:
          github-token: ${{ secrets.GITHUB_TOKEN }}
          path-to-lcov: ./coverage/lcov.info
```

**Comment by afinetooth** [comment]
Hi @peppescg.

OK, just FYI, that commit is not the latest commit, per [commit history](https://github.com/coverallsapp/github-action/commits/main/) / [releases](https://github.com/coverallsapp/github-action/releases), and the version you're pointing to with that commit is actually `v2.3.1` and not `v2.3.0` (per your comment).

Also, per this config:

```
 name: Send coverage to Coveralls
        uses: coverallsapp/github-action@1134c89e4bf10443bf8f0ec69640b667cfd91041 #v2.3.0
        with:
          github-token: ${{ secrets.GITHUB_TOKEN }}
          path-to-lcov: ./coverage/lcov.info
```

You're using an outdated input option, `path-to-lcov`, which was retired with `@v1` in lieu of [`file` (or `files`)](https://github.com/marketplace/actions/coveralls-github-action#:~:text=.%20More%20Info-,file,your%20test%20suite.%20Example%3A%20files%3A%20coverage/test1.lcov%20coverage/test2.lcov,-format).

So the first thing to do is try that.

I'd also recommend just calling the action with `@v2` which will grab the latest release in the `2.x.x` series, so:

```
 name: Send coverage to Coveralls
        uses: coverallsapp/github-action@v2
        with:
          github-token: ${{ secrets.GITHUB_TOKEN }}
          file: ./coverage/lcov.info
```

If you're still having an issue, please share the response from your CI build log.

Also, if you are meaning to run the underlying integration that the github action installs, it's no longer `node-coveralls` (aka. `coveralls npm package`), it's now a cross-platform binary called the [Universal Coverage Reporter](https://docs.coveralls.io/integrations#universal-coverage-reporter), which you can find here: https://github.com/coverallsapp/coverage-reporter

Usage instructions for coverage-reporter are here:
https://github.com/coverallsapp/coverage-reporter?tab=readme-ov-file#usage

Per that, the equivalent command, once you're sure coverage-reporter has been successfully downloaded and installed in your runner environment, is:

```
coveralls report coverage/lcov.info
```

Or possibly:

```
./coveralls report coverage/lcov.info
```

And you'l probably need to include your `repo_token`:

```
./coveralls report coverage/lcov.info --repo-token=<your_repo_token>
```

**Comment by peppescg** [comment]
@afinetooth ooh thanks a lot. My fault I also looked at the doc, but because I didn't see any error in the job I thought it was ok. Let me update the GH action.
Thanks!

About using @v2 or in general using the version rather than the commit, there is a known supply chain security issue based on my decision. Using a sha avoid to download possible malicious package in your supply chain. For reference I recommend this 
https://stacklok.com/blog/automating-security-for-github-actions-in-minder

**Comment by peppescg** [comment]
@afinetooth I continue to see unknown in coveralls.io and I cannot see the commen in my Pr 🤔 

<img width="685" alt="Screenshot 2024-10-25 at 21 53 44" src="https://github.com/user-attachments/assets/08ece3c4-d986-4f7e-be10-0a00aec7e9c4">

**Comment by afinetooth** [comment]
@peppescg To answer your questions:

**tl:dr**:
> Does I need to have it in main before it works in the next PR

Yes.

**Details**:
Yes, sorry. I think of [Notifications](https://docs.coveralls.io/app-notifications) (like [PR Comments](https://docs.coveralls.io/app-notifications#pull-request-comments)) as distinct functionality unrelated to our integrations, since they are features of our [Web App](https://docs.coveralls.io/app-ui)/Web Service at Coveralls.io.

Integrations get your coverage stats _into_ Coveralls.io, then Coveralls does the rest, like tracking your coverage stats, comparing them to previous builds and figuring out whether, and what, to send in [Status Updates/Checks](https://docs.coveralls.io/app-notifications#status-updates), or [PR Comments](https://docs.coveralls.io/app-notifications#pull-request-comments).

For all notifications that report on a _change_ in coverage, a "coverage diff" is required to generate that information, so Coveralls needs to be able to find a "previous build" (aka. "base build" when the build is a PR build).

So yes, if you're seeing "**unknown**" in your [badge](https://docs.coveralls.io/app-notifications#badges), that means Coveralls can't find the base build for your PR Build and therefore doesn't have half the coverage stats it needs to perform a coverage diff. 

So you'll want to merge your Coveralls configuration changes into your default branch so that you start uploading coverage reports to Coveralls for `push` builds on your default branch, and, after that, all future PRs _should_ be able to find their base builds, which are required to "do" a coverage diff and generate the content of a PR Comment.

**Comment by peppescg** [comment]
@afinetooth that makes sense, thanks a lot. I merged it.

**Comment by peppescg** [comment]
Sorry for bother you @afinetooth , but I have the job completed correctly and the coverage uploaded to coveralls, but I don't see the comments on PR. What am I missing 🤔 ?

<img width="791" alt="Screenshot 2024-10-26 at 00 46 07" src="https://github.com/user-attachments/assets/5b89fe28-6f02-4bca-a5cf-bb101f462128">

<img width="1170" alt="Screenshot 2024-10-26 at 00 49 55" src="https://github.com/user-attachments/assets/45dd9a6a-6e54-4de3-ba70-45d2a6265a50">

_____
Let me summarize what I have:

- GH action on main, and on PR triggered after a PR merged that run test and coveralls  
- enabled PR comments on coveralls settings
- coverage uploaded to coveralls

**Comment by afinetooth** [comment]
@peppescg looks like progress. That's good!

I will need to look at your project to determine which of several causes may be at play. Can you please share the Coveralls URL for your project? Even better would be the Coveralls URL for an example build, or builds, that aren't receiving PR Comments, for which you're expecting them. 

I will try to find your project in the meantime...

**Comment by afinetooth** [comment]
OK, I think I found your project, and all look sgood, but I suspect you did not add the [Coveralls Commenter Bot](https://github.com/coveralls) to your repo. Could that be it? Here are the [instructions](https://docs.coveralls.io/app-notifications#2-add-the-coveralls-bot-to-your-repo-at-github).

You are on our legacy integration with GitHub, which is based on an OAuth App, which must leverage _some_ user token in order to complete requests to the GitHub API. To leave PR Comments, our [Commenter Bot](https://github.com/coveralls) is the user that needs the access to your repo in order to leave the comment.

If you are concerned about the security of this, or the cost of adding the Commenter Bot as a user of your org in order to limit its access, I recommend migrating to our [New GitHub Integration](https://coveralls.io/new-github-integration-beta), which is based on a GitHub App and can use its own access token to make requests to GitHub. As a result, you won't need to user an external user to receive PR Comments, or worry about the token status of your repo "owner" etc.

To migrate:

1. Go here: https://coveralls.io/new-github-integration-beta.
2. Click JOIN NEW GITHUB INTEGRATION BETA.
3. Share your GitHub Org and the name(s) of your [Migration Lead(s)](https://coveralls.io/new-github-integration-beta#migration-leads).

**Comment by peppescg** [comment]
@afinetooth I need to pay a beer to you. Now it works, amazing. Thanks for your time.

**Comment by afinetooth** [comment]
LOL. 🍺 🍺 Cheers to you @peppescg. You're the one that got it working! 🍻
